src/Internal/GraphQL/BaseQueryType.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Automattic\WooCommerce\Internal\GraphQL;
+
+use GraphQL\Type\Definition\ObjectType;
+use GraphQL\Type\Definition\ResolveInfo;
+
+/**
+ * Base class for query types.
+ *
+ * To create a new query:
+ *
+ * 1. Add a new class inheriting this one in the "QueryTypes" folder/namespace.
+ * 2. Add the class name to "get_object_type_classes" in "RootQueryType".
+ * 3. Add the class name to "$provides" in "GraphqlTypesServiceProvider"
+ *    so that it can be resolved with the dependency injection container.
+ */
+abstract class BaseQueryType extends ObjectType {
+
+	/**
+	 * An instance of the dependency injection container.
+	 * Can be used by derived classes.
+	 *
+	 * @var \Psr\Container\ContainerInterface
+	 */
+	protected $container;
```

Comment:
I wonder if instead we should rely on dependency injection?

Reply:
Ideally, yes, and that was the first thing I tried. However there's a problem: the classes for defining query types in the `php-graphql` framework need to initialize in the constructor, and sometimes this involves accessing dependencies, but the `init` method hasn't been invoked yet.